site/src/pages/tutorials/thrift/blog/define-service-methods.mdx

```diff
@@ -0,0 +1,185 @@
+---
+menuTitle: Define service methods
+order: 3
+type: step
+level: basic
+---
+
+# Defining service methods
+
+Previously, we ran a service on a server and also confirmed that our client could make a call to the service.
+In this step, we'll define service methods and other components including objects and an exception for server and client each.
+We'll leave the service methods empty for now to implement one by one in later steps of this tutorial.
+
+<TutorialSteps current={3} />
+
+## What you need
+
+You need to have the files obtained from previous steps:
+
+- blog.thrift
+- BlogServiceImpl.java
+- BlogClient.java
+
+
+## 1. Write the thrift file
+
+Let's update the `blog.thrift` file we created earlier in [Step 1. Create a service](/tutorials/thrift/blog/create-service).
+We'll add structs, an exception, and service methods.
+You can see the full version of the file [here](https://github.com/line/armeria-examples/blob/main/tutorials/thrift/src/main/thrift/blog.thrift).
+
+1. Add the `BlogPost` struct.
+  ```thrift filename=blog.thrift
+  /**
+   * An object which represents a blog post.
+   */
+  struct BlogPost {
+      1: i32 id;
+      2: string title;
+      3: string content;
+      4: i64 createdAt;
+      5: i64 modifiedAt;
+  }
+  ```
+2. Add structs for request and response objects as follows.
+  ```thrift filename=blog.thrift
+  /**
+   * A request object for creating a blog post.
+   */
+  struct CreateBlogPostRequest {
+      1: string title;
+      2: string content;
+  }
```

Comment:
> generally we prefer two spaces in our documentation for code (to avoid lines getting too long)

I see, will do!

> I'm not too strong on this , but I think it's fine to remove javadocs for the tutorial.

Will do! (I thought they were meant to be added to the tutorial 😅)